### PR TITLE
Run test suite as non-root

### PR DESCRIPTION
**Pull request by dcermak**

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -348,9 +348,17 @@ jobs:
...
-        run: sudo --preserve-env -H tox -e ${{ matrix.toxenv }} -- -n 3 --reruns 3
+      - name: Run the tests
+        run: tox -e ${{ matrix.toxenv }} -- -n auto --reruns 3 --pytest-container-log-level=debug
         env:
           CONTAINER_RUNTIME: ${{ matrix.container_runtime }}
           OS_VERSION: ${{ matrix.os_version }}
           TARGET: ${{ matrix.testing_target != '' && matrix.testing_target || 'obs' }}
+
+      - name: Run tests as root for podman
+        run: sudo --preserve-env tox -e ${{ matrix.toxenv }} -- -n auto --reruns 3 --pytest-container-log-level=debug
```
Comment (dirkmueller) [inline .github/workflows/ci.yaml:359]: why did you remove the "-H" parameter which is needed to set the home directory so that pip caching works?

Comment (dcermak) [reply .github/workflows/ci.yaml]: that was a mistake

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -348,9 +348,17 @@ jobs:
...
-        run: sudo --preserve-env -H tox -e ${{ matrix.toxenv }} -- -n 3 --reruns 3
+      - name: Run the tests
+        run: tox -e ${{ matrix.toxenv }} -- -n auto --reruns 3 --pytest-container-log-level=debug
         env:
           CONTAINER_RUNTIME: ${{ matrix.container_runtime }}
           OS_VERSION: ${{ matrix.os_version }}
           TARGET: ${{ matrix.testing_target != '' && matrix.testing_target || 'obs' }}
+
+      - name: Run tests as root for podman
+        run: sudo --preserve-env -H tox -e ${{ matrix.toxenv }} -- -n auto --reruns 3 --pytest-container-log-level=debug
```
Comment (dirkmueller) [inline .github/workflows/ci.yaml:359]: the `-n 3` was intentional because `-n auto` triggers too many parallel pulls which is hitting rate limiting of the CDN and then causes testcases to fail

**Review by dirkmueller — DISMISSED** [review]

File: conftest.py [hunk 3]
```diff
@@ -63,6 +63,17 @@ def container_git_clone(request: SubRequest, tmp_path):
...
     )
+    # fix file permissions for the git copied git repo
+    check_output(
+        [
+            runtime.runner_binary,
+            "exec",
+            container_fixture.container_id,
+            "/bin/sh",
+            "-c",
+            f"chown --recursive $(id -u):$(id -g) {git_repo_build.repo_name}",
```
Comment (dirkmueller) [inline conftest.py:74]: there gotta be a better way than this... is the issue that we'r enot setting a home directory on the container?

File: tests/test_go.py [hunk 4]
```diff
@@ -67,19 +67,22 @@ def test_build_kured(auto_container_per_test, container_git_clone):
     [
         GitRepositoryBuild(
             repository_url="https://github.com/helm/helm.git",
-            repository_tag="v3.14.2",
-            build_command="make build test-unit",
+            repository_tag="v3.14.3",
+            build_command="make build && make test-unit",
```
Comment (dirkmueller) [inline tests/test_go.py:71]: where's the difference between the two? as we don't run with "-j" it should be the same?

File: conftest.py (hunk 3 above)
Comment (dcermak) [reply conftest.py:74]: No the issue is that `docker cp ` operates in archive mode by default preserving the uid & gid of the git repository. So the cloned repo has everything owned by uid 1000 (well, on my box, but generally != 0) and the process in the container is run under uid 0. And with that anything invoking git on that repo fails due to an ownership missmatch.

File: tests/test_go.py (hunk 4 above)
Comment (dcermak) [reply tests/test_go.py:71]: actually none, I thought it would fix the test issues before looking closer

**Review by dirkmueller — APPROVED** [review]
